scripts/declsIn.lean

```diff
@@ -0,0 +1,27 @@
+import Std
+import Mathlib.Lean.Expr.Basic
+import Mathlib.Data.BinaryHeap
+import Mathlib.Lean.CoreM
+
+open Lean
+
+/-- Given an array of modules, this function returns the sorted list of all non-blacklisted
+declarations provided by those modules.
+
+The script `decl-name-changes.sh` wraps this and uses `git` to detect which files have changed.
+Then it computes the `diff` between the arrays returned by this function.  In general, users should
+prefer that instead of using this script directly. -/
+def getDeclsInPackagesNoBlacklist (pkgs : Array Name) : CoreM (Array Name) := do
+  let arr_decls ← pkgs.mapM fun pkg => do
+    (← Std.Tactic.Lint.getDeclsInPackage pkg).filterM (Functor.map (! ·) ∘ Lean.Name.isBlackListed)
+  return arr_decls.flatten.heapSort (toString · < toString ·) |>.deduplicateSorted
```

Comment:
More out of curiosity but why heapsort? and this seems to do a lot of conversions to string all the time why not use `Lean.Name.lt`